addons/addon-base-raas/packages/base-raas-services/lib/plugins/env-provisioning-plugin.js

```diff
@@ -384,12 +407,20 @@ async function rstudioCleanup(requestContext, updatedEnvironment, container) {
   let connectionTypeValue;
   if (connectionType) {
     connectionTypeValue = connectionType.OutputValue;
-    if (connectionTypeValue.toLowerCase() === 'rstudio') {
-      const dnsName = _.find(updatedEnvironment.outputs, x => x.OutputKey === 'Ec2WorkspaceDnsName').OutputValue;
+    if (connectionTypeValue.toLowerCase() === 'rstudio' || connectionTypeValue.toLowerCase() === 'rstudiov2') {
+      let dnsName;
+      if (connectionTypeValue.toLowerCase() === 'rstudio') {
+        dnsName = _.find(updatedEnvironment.outputs, x => x.OutputKey === 'Ec2WorkspaceDnsName').OutputValue;
+        //Moved the code here to handle deletion of RstudioV1, This will be changed when termination
+        // code for RstudioV2 comes in. Please ignore this on code review
+        const environmentDnsService = await container.find('environmentDnsService');
+        await environmentDnsService.deleteRecord('rstudio', updatedEnvironment.id, dnsName);
+      }
+      else if (connectionTypeValue.toLowerCase() === 'rstudiov2') {
+        // TODO: Add code to remove record for new Rstudio version
```

Comment:
Will this be addressed in the current PR?

Reply:
This will be part of the terminate workflow. So we will address this in the respective PR raised for the termination workflow.